plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/DeltaLakeSchemaSupport.java

```diff
@@ -198,13 +198,15 @@ public static ColumnMappingMode getColumnMappingMode(MetadataEntry metadata, Pro
                 return ColumnMappingMode.NAME;
             }
 
-            boolean supportsColumnMappingReader = protocolEntry.readerFeaturesContains(COLUMN_MAPPING_FEATURE_NAME);
-            boolean supportsColumnMappingWriter = protocolEntry.writerFeaturesContains(COLUMN_MAPPING_FEATURE_NAME);
-            checkArgument(
-                    supportsColumnMappingReader == supportsColumnMappingWriter,
-                    "Both reader and writer features must have the same value for 'columnMapping'. reader: %s, writer: %s", supportsColumnMappingReader, supportsColumnMappingWriter);
-            if (!supportsColumnMappingReader) {
-                return ColumnMappingMode.NONE;
+            if (protocolEntry.supportsReaderFeatures() && protocolEntry.supportsWriterFeatures()) {
```

Comment:
This is required because Delta protocol allows a combination of minReaderVersion:2 and minWriterVersion:7.